CI: kvstoremesh unit test ClusterMeshServicesTestSuite/TestRemoteServiceObserver panic

## CI failure

In an [unrelated PR](https://github.com/cilium/cilium/pull/32168), I saw a [panic](https://github.com/cilium/cilium/actions/runs/8820644829/job/24214722229) in a kvstoremesh privileged unit test:

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓                                           
┃   PANIC  package: github.com/cilium/cilium/pkg/clustermesh • Test/ClusterMeshServicesTestSuite/TestRemoteServiceObserver   ┃                                           
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛                                           
panic: send on closed channel                                                                                                                                            
                                                                                                                                                                         
goroutine 1132 [running]:                                                                                                                                                
github.com/cilium/cilium/pkg/clustermesh/common.(*remoteCluster).getClusterConfig.func1({0x2931a90, 0x400101a5f0})                                                       
	/home/runner/work/cilium/cilium/pkg/clustermesh/common/remote_cluster.go:293 +0xe0                                                                                       
github.com/cilium/cilium/pkg/controller.(*controller).runController(0x400074f2c0, {{{0x25638fb, 0x1a}}, {0x0, 0x0}, 0x4001453230, 0x0, 0x2690a18, 0x0, 0x6fc23ac00, ...})
	/home/runner/work/cilium/cilium/pkg/controller/controller.go:254 +0x108                                                                                                  
created by github.com/cilium/cilium/pkg/controller.(*Manager).createControllerLocked in goroutine 1065                                                                   
	/home/runner/work/cilium/cilium/pkg/controller/manager.go:111 +0x3d0                                                                                                     
FAIL	github.com/cilium/cilium/pkg/clustermesh	0.935s                                                                                                                       
                                                                                                                                                                         
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/pkg/clustermesh/kvstoremesh   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

--- FAIL: TestRemoteClusterRun (120.00s)

time="2024-04-24T17:34:09Z" level=info msg="Creating etcd client" ConfigPath= KeepAliveHeartbeat=15s KeepAliveTimeout=25s ListLimit=256 MaxInflight=100 RateLimit=100 subsys=kvstore
time="2024-04-24T17:34:09Z" level=info msg="Connecting to etcd server..." config= endpoints="[http://127.0.0.1:4002/]" subsys=kvstore
time="2024-04-24T17:34:09Z" level=info msg="New lease successfully acquired" LeaseID=5326789763116035469 TTL=25s config= endpoints="[http://127.0.0.1:4002/]" subsys=kvstore
time="2024-04-24T17:34:09Z" level=info msg="Got lock lease ID 49ec8f1127b7ed8d" config= endpoints="[http://127.0.0.1:4002/]" subsys=kvstore
time="2024-04-24T17:34:09Z" level=info msg="Initial etcd session established" config= endpoints="[http://127.0.0.1:4002/]" subsys=kvstore
time="2024-04-24T17:34:09Z" level=info msg="New lease successfully acquired" LeaseID=5326789763116035478 TTL=15m0s config= endpoints="[http://127.0.0.1:4002/]" subsys=kvstore
    dummy.go:84: Timed out waiting to acquire the kvstore lock
```

--- a/pkg/clustermesh/common/remote_cluster.go
+++ b/pkg/clustermesh/common/remote_cluster.go
@@ -394,7 +394,7 @@ func (rc *remoteCluster) onInsert() {
 // In this case, we don't want to drain the known entries, otherwise
 // we would break existing connections when the agent gets restarted.
 func (rc *remoteCluster) onStop() {
-	rc.controllers.RemoveAllAndWait()
+	_ = rc.controllers.RemoveControllerAndWait(rc.remoteConnectionControllerName)
 	close(rc.changed)
 	rc.Stop()
 }

